src/interface/desktop/main.js

```diff
@@ -428,162 +428,211 @@ function addCSPHeaderToSession () {
+     shortcutWin.setHasShadow(true);
+     shortcutWin.setVibrancy('popover');
+
+     shortcutWin.loadFile(tab);
+     shortcutWin.once('ready-to-show', () => {
+         shortcutWin.show();
+     });
+
+    shortcutWin.on('closed', () => {
+        shortcutWin = null;
     });
 
-    job.start();
-
-    win.loadFile(tab)
-
-    if (firstRun === true) {
-        firstRun = false;
-
-        // Create splash screen
-        let splash = new BrowserWindow({width: 400, height: 400, transparent: true, frame: false, alwaysOnTop: true});
-        splash.setOpacity(1.0);
-        splash.setBackgroundColor('#d16b4e');
-        splash.loadFile('splash.html');
-
-        // Show splash screen on app load
-        win.once('ready-to-show', () => {
-            setTimeout(function(){ splash.close(); win.show(); }, 4500);
+     return shortcutWin;
+ };
+ app.whenReady().then(() => {
+     addCSPHeaderToSession();
+
+     ipcMain.on('set-title', handleSetTitle);
+     ipcMain.handle('handleFileOpen', (event, type) => handleFileOpen(type));
+     ipcMain.on('update-state', (event, arg) => { console.log(arg); event.reply('update-state', arg); });
+     ipcMain.on('needsSubscription', (event, arg) => { console.log(arg); event.reply('needsSubscription', arg); });
+     ipcMain.on('navigate', (event, page) => { win.loadFile(page); });
+     ipcMain.on('navigateToWebApp', (event, page) => { shell.openExternal(`${store.get('hostURL')}/${page}`); });
+
+     ipcMain.handle('getFiles', getFiles);
+     ipcMain.handle('getFolders', getFolders);
+     ipcMain.handle('removeFile', removeFile);
+     ipcMain.handle('removeFolder', removeFolder);
+     ipcMain.handle('setURL', setURL);
+     ipcMain.handle('getURL', getURL);
+     ipcMain.handle('setToken', setToken);
+     ipcMain.handle('getToken', getToken);
+     ipcMain.handle('getUserInfo', getUserInfo);
+     ipcMain.handle('syncData', (event, regenerate) => syncData(regenerate));
+     ipcMain.handle('deleteAllFiles', deleteAllFiles);
+
+     createWindow();
+
+     app.setAboutPanelOptions({
+         applicationName: "Khoj",
+         applicationVersion: khojPackage.version,
+         version: khojPackage.version,
+         authors: "Saba Imran, Debanjum Singh Solanky and contributors",
+         website: "https://khoj.dev",
+         copyright: "GPL v3",
+         iconPath: path.join(__dirname, 'assets', 'icons', 'favicon-128x128.png')
+     });
+
+     app.on('ready', async() => {
+         try {
+             const result = await todesktop.autoUpdater.checkForUpdates();
+             if (result.updateInfo) {
+                 console.log("Desktop app update found:", result.updateInfo.version);
+                 todesktop.autoUpdater.restartAndInstall();
+             }
+         } catch (e) {
+             console.warn("Desktop app update check failed:", e);
+         }
+     });
+     openShortcut = false;
+     globalShortcut.register('CommandOrControl+Shift+K', () => {
```

Comment:
Nice! Small nit: maybe this can go in a separate variable. Makes the code a bit more easy to understand on global constructs. Kind of like [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)). Something like:
```
const openShortcutWindowKeybidning = 'CommandOrControl+Shift+K';
```